unicodetools/src/main/java/org/unicode/props/PropertyParsingInfo.java

```diff
@@ -1510,6 +1536,38 @@ private static void parseFields(
                         value = "No";
                     }
                 }
+                if ((propInfo.property == UcdProperty.Math_Entity_Name
+                                || propInfo.property == UcdProperty.Math_Entity_Set)
+                        && indexUnicodeProperties.ucdVersion.compareTo(Utility.UTR25_REVISION_16)
+                                < 0) {
+                    merger = IndexUnicodeProperties.MULTIVALUED_JOINER;
+                }
+                if (propInfo.property == UcdProperty.Math_Descriptive_Comments
+                        && indexUnicodeProperties.ucdVersion.compareTo(Utility.UTR25_REVISION_16)
+                                < 0) {
+                    merger = new PropertyUtilities.NullIgnorer();
+                }
+                if (propInfo.property == UcdProperty.Math_Class_Ex
+                        && indexUnicodeProperties.ucdVersion.compareTo(Utility.UTR25_REVISION_16)
+                                < 0) {
+                    merger = new PropertyUtilities.RedundancyIgnorer();
+                }
+                if (propInfo.property == UcdProperty.Math_Class_Ex
+                        && indexUnicodeProperties.ucdVersion.compareTo(VersionInfo.UNICODE_6_0) < 0
```

Comment:
Is it worth adding a comment explaining why you are doing this test?
Also, if you are proposing that we *should* support multiple values (e.g., exclamation mark), will we need a broader support for this type of construction?

Reply:
If we actually want multivalued MathClass (which I am starting to think we might), this will all get a lot simpler (because it will be handled like the other multivalued Math properties, with a MultivaluedJoiner).

I will copy the comment from Math_Class for now.